expression/builtin_cast.go

```diff
@@ -570,6 +570,9 @@ func (b *builtinCastIntAsDurationSig) Clone() builtinFunc {
 }
 
 func (b *builtinCastIntAsDurationSig) evalDuration(row chunk.Row) (res types.Duration, isNull bool, err error) {
+	if b.tp.Decimal > types.MaxFsp {
+		return res, true, errTooBigPrecision.GenWithStackByArgs(b.tp.Decimal, "CAST", types.MaxFsp)
```

Comment:
isNull should be false on this error case?

```suggestion
		return res, false, errTooBigPrecision.GenWithStackByArgs(b.tp.Decimal, "CAST", types.MaxFsp)
```
is better?